cv-19-i-need-help/Boundary/V2/Handler.cs

```diff
@@ -12,105 +12,150 @@ namespace CV19INeedHelp.Boundary.V2
+            var getRequestsUseCase = new CV19INeedHelp.UseCases.V2.GetHelpRequestsUseCase(getRequestGateway);
+
+            var master = QueryValueGiven("master", request) && bool.TryParse(request.QueryStringParameters["master"], out var v)
+                ? SetValue(v, "master")
+                : ValueNotSet<bool>("master");
+            var uprn = GetQueryParameter("uprn", request);
+            var postcode = GetQueryParameter("postcode", request);
+            var address = GetQueryParameter("address", request);
+            var firstName = GetQueryParameter("first_name", request);
+            var lastName = GetQueryParameter("last_name", request);
+
+            try
+            {
+                var requests = new ResidentSupportAnnexResponseList
+                {
+                    HelpRequests = getRequestsUseCase
+                        .GetHelpRequests(uprn, postcode, address, firstName, lastName, master).ToResponse(),
+                };
+                var resp = ConvertToCamelCasedJson(requests);
+                LambdaLogger.Log("Records retrieval success: " + resp);
+
+                return new APIGatewayProxyResponse
+                {
+                    IsBase64Encoded = true,
+                    StatusCode = 200,
+                    Body = resp
+                };
+            }
+            catch (InvalidQueryParameter e)
+            {
+                return new APIGatewayProxyResponse
+                {
+                    IsBase64Encoded = true,
+                    StatusCode = 400,
+                    Body = "Error processing request: " + e.Message
+                };
+            }
+            catch (Exception e)
+            {
+                return SendServerErrorResponse(e);
+            }
+        }
+
+        private static APIGatewayProxyResponse SendServerErrorResponse(Exception e)
+        {
+            LambdaLogger.Log("Error: " + e.Message);
+            return new APIGatewayProxyResponse
+            {
+                IsBase64Encoded = true,
+                StatusCode = 500,
+                Body = "Error processing request: " + ". Error Details: " + e.Message + e.StackTrace
+            };
+        }
+
+        private static string GetQueryParameter(string keyName, APIGatewayProxyRequest request)
+        {
+            return QueryValueGiven(keyName, request)
+                ? SetValue(request.QueryStringParameters[keyName], keyName)
+                : ValueNotSet<string>(keyName);
+        }
+
+        public APIGatewayProxyResponse GetHelpRequest(APIGatewayProxyRequest request, ILambdaContext context)
+        {
+            var getRequestGateway = new CV19INeedHelp.Gateways.V1.INeedHelpGateway(new Cv19SupportDbContext(_connectionString));
+            var getRequestUseCase = new CV19INeedHelp.UseCases.V1.GetHelpRequestUseCase(getRequestGateway);
+            var requestParams = request.PathParameters;
+
+            var requestedId = Convert.ToInt32(requestParams["id"]);
```

Comment:
This could result in an error. Maybe we should be returning a 400 if the id isn't numerical? Definitely work for a different PR though